### PR TITLE
[12.0][ADD] helpdesk_mgmt_timesheet_addline

### PR DESCRIPTION
**Pull request by ChristianSantamaria**
Editing in the upper part of the tree is done, but I couldn't find a way to open the tree to add a new timesheet, so I made a button that open a form and when saving it is automatically added to the tree without needing to refresh the page

### CODE REVIEW

**Comment by lmiguens-solvos** [comment]
I tested it and it works.
Although the button is very close to the grid and the translation is missing.
I suppose waiting to see if it is finally done like this or not, right?

**Comment by ChristianSantamaria** [comment]
Yes, with the design and the translations I'm waiting, since maybe David suggests another way to do it

File: helpdesk_mgmt_timesheet_addline/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "summary": """
+
+    """,
+    "version": "12.0.1.0.0",
+    "license": "LGPL-3",
+    "category": "After-Sales",
+    "author": "Solvos",
+    "website": "https://github.com/solvosci/slv-helpdesk",
+    "depends": [
+        "hr_timesheet",
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_addline/__manifest__.py]: Wrong dependency:
```suggestion
        "helpdesk_mgmt_timesheet",
```

File: helpdesk_mgmt_timesheet_addline/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    ],
+    "data": [
+        "views/helpdesk_ticket_timesheet_views.xml",
```
Comment (dalonsod) [inline helpdesk_mgmt_timesheet_addline/__manifest__.py:18]: Please split in two files (one for each affected model)

**Review by dalonsod — CHANGES_REQUESTED** [review]
Only code review. When required changes attended, I'll test it

**Comment by dalonsod** [comment]
Locally tested, LGTM.

Anyway, updated with these improvements:
- Better definition of button linked action and its form view.
- Styles changed for some of the buttons.
- TODO comment for dummy action method.
- Switched fields on new form view, in order to prevent annoying datepicker for timesheet date.
- Added description to readme and manifest file.

@ChristianSantamaria could you test it, update `README` file and squash commits?

**Comment by ChristianSantamaria** [comment]
LGTM

**Review by dalonsod — APPROVED** [review]
LGTM

**Comment by dalonsod** [comment]
@lmiguens-solvos anything else, or is it OK for you? :)

**Comment by lmiguens-solvos** [comment]
> @lmiguens-solvos anything else, or is it OK for you? :)

When I updated the addon give me this error: 
"psycopg2.IntegrityError: llave duplicada viola restricción de unicidad «act_window_view_unique_mode_per_action»
DETAIL:  Ya existe la llave (act_window_id, view_mode)=(190, form)."
Then I uninstalled it and reinstalled it. And then it works.

I tried the addon and everything was fine, although it would leave the editable attibute at bottom instead of top, since when click the "Add line" I think it is better to open that new line next to where I clicked  "Add line".

On the other hand I verified that the operation of the new button is different from the link "Add line" in the sense that in the window that opens with the new button when clicking "Save" the record is already in the database, although it is discarded later the edition of the ticket, which does not happen with the line that is opened from "Add line"... 
But hey, I do not put problems to that. It is just a comment.

**Comment by dalonsod** [comment]
> When I updated the addon give me this error:
> "psycopg2.IntegrityError: llave duplicada viola restricción de unicidad «act_window_view_unique_mode_per_action»
> DETAIL: Ya existe la llave (act_window_id, view_mode)=(190, form)."
> Then I uninstalled it and reinstalled it. And then it works.
> 

I think this error should no longer occur, since action and view definition is actually changed in the last commit in order to provent it. Could you update again the addon and check if no error is fired?

> I tried the addon and everything was fine, although it would leave the editable attibute at bottom instead of top, since when click the "Add line" I think it is better to open that new line next to where I clicked "Add line".

Looks coherent. I'll change it.
 
> On the other hand I verified that the operation of the new button is different from the link "Add line" in the sense that in the window that opens with the new button when clicking "Save" the record is already in the database, although it is discarded later the edition of the ticket, which does not happen with the line that is opened from "Add line"...
> But hey, I do not put problems to that. It is just a comment.

Yes, at this time this is the only way we know to achieve the feature, and it's not blocking. I'll add it to `ROADMAP` section.

**Comment by dalonsod** [comment]
@lmiguens-solvos comments attended

**Comment by lmiguens-solvos** [comment]
> @lmiguens-solvos comments attended

ok. Seen and ok

**Review by lmiguens-solvos — APPROVED** [review]
LGTM

**Pull request merged**